### ISSUE TITLE
[bug] private requirement being treated as a build_requirement instead

### ISSUE THREAD

**Issue by ericriff** [conan-io/conan#8923]
In this PR https://github.com/conan-io/conan-center-index/pull/5432 I'm trying to use `linux-syscall-support` as a private `requirement`.
```
    def requirements(self):
        if self.settings.os == "Linux":
            self.requires("linux-syscall-support/cci.20200813", private=True)
         if self.settings.os in ["Linux", "Android"]:
            self.requires("openssl/1.1.1k")
        self.requires("zlib/1.2.11")
```

If I try to crosscompile using 2 profiles the build fails reporting it doesn't find the `Findlinux-syscall-support.cmake` module. Inspecting the logs they show the following, which really confuses me:

```
Requirements
    getsentry-crashpad/cci.20210420 from local cache - Cache
    linux-syscall-support/cci.20200813 from 'conan-center' - Downloaded
    openssl/1.1.1k from 'conan-center' - Downloaded
    zlib/1.2.11 from 'conan-center' - Downloaded
Packages
    getsentry-crashpad/cci.20210420:fb2c5ec7c518545ec30e1f9f2e00f4db1368d5b5 - Build
    linux-syscall-support/cci.20200813:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Download
    openssl/1.1.1k:19729b9559f3ae196cad45cb2b97468ccb75dcd1 - Download
    zlib/1.2.11:19729b9559f3ae196cad45cb2b97468ccb75dcd1 - Download

```

But then
```
getsentry-crashpad/cci.20210420: Applying build-requirement: linux-syscall-support/cci.20200813
```

And no `FindPKG` is generated for linux-syscall-support:
```
getsentry-crashpad/cci.20210420@asd/asd: Copying sources to build folder
getsentry-crashpad/cci.20210420@asd/asd: Building your package in /home/eric/.conan/data/getsentry-crashpad/cci.20210420/asd/asd/build/ba86582b2b2b136114454d185bd1421ebb978603
getsentry-crashpad/cci.20210420@asd/asd: Generator cmake_find_package created FindOpenSSL.cmake
getsentry-crashpad/cci.20210420@asd/asd: Generator cmake_find_package created FindZLIB.cmake
getsentry-crashpad/cci.20210420@asd/asd: Generator cmake created conanbuildinfo.cmake
```

Getting rid of the `Private=True` fixes the issue for me.
Is this intended behavior? 

### Environment Details (include every applicable attribute)
  * Operating System+version: Ubuntu 18.04
  * Compiler+version:
  * Conan version: 1.32.1 (Also tested 1.36)
  * Python version: 3.6

**Comment by danimtb** [comment]
Private requirements have a weir behavior and the general advice is to avoid them whenever it's possible. This is something we really want to improve in Conan 2.0 with a new graph model that clearly defines the behavior of different types of dependencies.

I took a look at the example and I am not sure why the requirement should be private. Maybe declaring it as nonprivate is enough for your case?

**Comment by ericriff** [comment]
This package only provides a header used during build. I didn't want it to propagate as a transitive dependency when my package gets consumed, sok I made it private.

I tried making it a build_requirement instead, but then I don't get the findPackage.cmake module so I can't consume it.

**Comment by danimtb** [comment]
totally understand your use case and this kind of relationships is what we'd like to get fixed in the graph model for conan 2.0 as we think is crucial to achieve the right definition of dependencies. Meanwhile, I'd suggest you to declare it as a normal requirement to make it work. I know it is not ideal but is the best approach right now. Hope that helped to clarify the situation. Thank you!

**Comment by memsharded** [comment]
Closing this as solved by Conan 2.0 with the new requirement traits.
